kernel/src/scan/mod.rs

```diff
@@ -355,31 +381,48 @@ fn parse_partition_value(raw: Option<&String>, data_type: &DataType) -> DeltaRes
 /// Get the state needed to process a scan. In particular this returns a triple of
 /// (all_fields_in_query, fields_to_read_from_parquet, have_partition_cols) where:
 /// - all_fields_in_query - all fields in the query as [`ColumnType`] enums
-/// - fields_to_read_from_parquet - Which fields should be read from the raw parquet files
+/// - fields_to_read_from_parquet - Which fields should be read from the raw parquet files. This takes
+///   into account column mapping
 /// - have_partition_cols - boolean indicating if we have partition columns in this query
-fn get_state_info<'a>(
-    schema: &'a Schema,
+fn get_state_info(
+    schema: &Schema,
     partition_columns: &[String],
-) -> (Vec<ColumnType<'a>>, Vec<StructField>, bool) {
+    column_mapping_mode: &ColumnMappingMode,
+) -> (Vec<ColumnType>, Vec<StructField>, bool) {
     let mut have_partition_cols = false;
     let mut read_fields = Vec::with_capacity(schema.fields.len());
     // Loop over all selected fields and note if they are columns that will be read from the
     // parquet file ([`ColumnType::Selected`]) or if they are partition columns and will need to
     // be filled in by evaluating an expression ([`ColumnType::Partition`])
     let column_types = schema
         .fields()
-        .map(|field| {
+        .enumerate()
+        .map(|(index, field)| {
             if partition_columns.contains(field.name()) {
                 // todo: this is slow(ish)
                 // Store the raw field, we will turn it into an expression in the inner loop
                 // since the expression could be different for each add file
                 have_partition_cols = true;
-                ColumnType::Partition(field)
+                ColumnType::Partition(index)
             } else {
                 // Add to read schema, store field so we can build a `Column` expression later
                 // if needed (i.e. if we have partition columns)
-                read_fields.push(field.clone());
-                ColumnType::Selected(field)
+
+                match column_mapping_mode {
+                    ColumnMappingMode::Id => unimplemented!("Don't support id mapping yet"),
+                    ColumnMappingMode::Name => {
+                        // this is the value we'll look for in the parquet, so we need to pass the
+                        // physical name
+                        let (physical_field, physical_name) =
+                            get_name_mapped_physical_field(field).unwrap();
```

Comment:
`unwrap`?

Reply:
fixed by having a result returned by this